aerospike-core/src/client.rs

```diff
@@ -632,34 +633,38 @@ impl Client {
     ///     Err(err) => println!("Failed to execute scan: {}", err),
     /// }
     /// ```
-    pub fn scan<T>(
+    ///
+    /// # Panics
+    /// Panics if the async block fails
+    pub async fn scan<T>(
         &self,
         policy: &ScanPolicy,
         namespace: &str,
         set_name: &str,
         bins: T,
     ) -> Result<Arc<Recordset>>
     where
-        T: Into<Bins>,
+        T: Into<Bins> + Send + Sync + 'static,
```

Comment:
Preferably, this `IntoBins` trait should allow a lazy iterator so that everything fit on the stack and no heavy allocation would be needed. Thoughts?